File: week-01/day4/challenges/algo-anagrams-i/js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {
 return string1.toLowerCase().replaceAll(' ', '').split('').sort().join('') === string2.toLowerCase().replaceAll(' ', '').split('').sort().join('')
};

exports.anagramsFor = function(word, listOfWords) {
  let answer = []
  for(let words of listOfWords){
    if (word.toLowerCase().replaceAll(' ', '').split('').sort().join('') === words.toLowerCase().replaceAll(' ', '').split('').sort().join('')){
      answer.push(words)
    }

  }
  return answer
};

File: week-01/day4/challenges/algo-anagrams-i/python/character_match.py
# Don't forget to run the tests (and create some of your own)

# Part 1
def is_character_match(string1, string2):
    str1 = sorted(string1.lower().replace(' ', ''))
    str2 = sorted(string2.lower().replace(' ', ''))
    return str1 == str2


# print(is_character_match('charm', 'march') == True)


# # Part 2
def anagrams_for(word, list_of_words):
    answer = []
    str1 = sorted(word.lower())
    for words in list_of_words:
        if str1 == sorted(words.lower()):
            answer.append(words)
    return answer


# list_of_words = ["threads", "trashed", "hardest", "hatreds", "hounds"]

# print(anagrams_for("threads", list_of_words) == [
#       "threads", "trashed", "hardest", "hatreds"])
